integration-tests/test_register.py

```diff
@@ -0,0 +1,47 @@
+import contextlib
+import pytest
+import logging
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_register(insights_client):
+    """
+    Test `insights-client --register` when the client is registered
+    """
+    response=insights_client.run('--register', check=False)
+    assert response.returncode == 0,\
+        "An application returns 0 after successful registration"
```

Comment:
the default behaviour of `run()` is with `check=True`, so it can be dropped from the arguments; also, the `returncode` assertion for it can be dropped too

Reply:
I'd like for a test to be more descriptive. At least when registering is the main point of the test. I want to describe behaviour of register task - and return code is a part of this behaviour.